### PR TITLE
Update to ncurses version 6.0

### LINKED ISSUE #14: Add 6.0 release
This would be a breaking release, but we should probably add it at some point. We need to make sure everything is pinned correctly before we do though.

### PR DESCRIPTION
**Pull request by jjhelmus**
closes #14

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -7,16 +7,19 @@ package:
...
-  md5: 8cb9c412e5f2d96bc6f459aa8c6282a1
+  md5: ee13d052e1ead260d7c28071f46eefb1
   patches:
     # http://lists.gnu.org/archive/html/bug-ncurses/2011-04/txtkWQqiQvcZe.txt
     - fix.patch
     # http://lists.gnu.org/archive/html/bug-ncurses/2015-01/msg00016.html
     - ncurses-5.9-gcc-5.patch
+    - clang.patch
+    # This patch is not needed with GCC 4.8.5
+    #- cxx1z-remove-dynamic-exception-specifications.patch
```
Comment (jakirkham) [inline recipe/meta.yaml]: Should we be adding this patch if we are not using it or should we add it when we switch compilers?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -7,16 +7,19 @@ package:
...
   fn: ncurses-{{ version }}.tar.gz
   url: http://ftp.gnu.org/pub/gnu/ncurses/ncurses-{{ version }}.tar.gz
-  md5: 8cb9c412e5f2d96bc6f459aa8c6282a1
+  md5: ee13d052e1ead260d7c28071f46eefb1
   patches:
     # http://lists.gnu.org/archive/html/bug-ncurses/2011-04/txtkWQqiQvcZe.txt
     - fix.patch
     # http://lists.gnu.org/archive/html/bug-ncurses/2015-01/msg00016.html
     - ncurses-5.9-gcc-5.patch
+    - clang.patch
```
Comment (jakirkham) [inline recipe/meta.yaml]: I don't really follow this patch. Don't those functions require pointers?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -7,16 +7,19 @@ package:
 source:
   fn: ncurses-{{ version }}.tar.gz
   url: http://ftp.gnu.org/pub/gnu/ncurses/ncurses-{{ version }}.tar.gz
-  md5: 8cb9c412e5f2d96bc6f459aa8c6282a1
+  md5: ee13d052e1ead260d7c28071f46eefb1
```
Comment (jakirkham) [inline recipe/meta.yaml]: Should we use a new checksum type?

File: recipe/meta.yaml [hunk 4]
```diff
@@ -96,19 +99,30 @@ test:
...
+    - pkg-config ncursesw --libs | grep "^-L${PREFIX}/lib .* -lncursesw$"
+    - pkg-config tinfow --libs | grep "^-L${PREFIX}/lib .* -ltinfow$"
+    - pkg-config ncurses --libs | grep "^-L${PREFIX}/lib .* -lncursesw$"
+    - pkg-config tinfo --libs | grep "^-L${PREFIX}/lib .* -ltinfow$"
 
     # Usage tests
     - bash run_test.sh    # [osx]
-    - python run_test.py  # [osx]
+    # Python test will not pass until it is recompiled against ncurses 6.0
+    #- python run_test.py  # [osx]
```
Comment (jakirkham) [inline recipe/meta.yaml]: We should just move this over to the `python` feedstock. It's a poor separation of concerns for us to test this here.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -96,19 +99,30 @@ test:
...
 
     # Test ncurses library arguments.
-    - pkg-config ncursesw --libs | grep "^-L${PREFIX}/lib -lncursesw -ltinfow$"
-    - pkg-config tinfow --libs | grep "^-L${PREFIX}/lib -ltinfow$"
-    - pkg-config ncurses --libs | grep "^-L${PREFIX}/lib -lncurses -ltinfo$"
-    - pkg-config tinfo --libs | grep "^-L${PREFIX}/lib -ltinfo$"
+    - pkg-config ncursesw --libs | grep "^-L${PREFIX}/lib .* -lncursesw$"
+    - pkg-config tinfow --libs | grep "^-L${PREFIX}/lib .* -ltinfow$"
+    - pkg-config ncurses --libs | grep "^-L${PREFIX}/lib .* -lncursesw$"
+    - pkg-config tinfo --libs | grep "^-L${PREFIX}/lib .* -ltinfow$"
```
Comment (jakirkham) [inline recipe/meta.yaml]: Why are dropping `tinfo` from the `ncurses` lib checks? Having both of these were important for bioconda users.

**Comment by jakirkham** [comment]
Tried to do a cursory look through while I have a few minutes here. Couple of questions. Will need to come back and look at this again when I have more time.

Have not had time to look at `build.sh`. Will need to look at that more carefully as it seems there were some sweeping changes there. Are we sure all of those are needed or can they be reapproached after doing the version update?

**Comment by ocefpaf** [comment]
@jjhelmus I know you are probably very busy at the moment but do you think you'll have time to come back to this PR? If not, any idea of what is happening on Travis-CI?

**Comment by ocefpaf** [comment]
@conda-forge-admin, please rerender

**Comment by isuruf** [comment]
> any idea of what is happening on Travis-CI?

Remove the `run_test.py` or rename it to something else.

**Comment by ocefpaf** [comment]
> Remove the run_test.py or rename it to something else.

Do you mean we do not need that test on OS X? Copying the error message here to make it easier for others to comment.

```
===== testing package: ncurses-6.0-0 =====
running run_test.py
Traceback (most recent call last):
  File "/Users/travis/miniconda3/conda-bld/ncurses_1508961917849/test_tmp/run_test.py", line 5, in <module>
    import curses
  File "/Users/travis/miniconda3/conda-bld/ncurses_1508961917849/_t_env/lib/python3.6/curses/__init__.py", line 13, in <module>
    from _curses import *
ImportError: dlopen(/Users/travis/miniconda3/conda-bld/ncurses_1508961917849/_t_env/lib/python3.6/lib-dynload/_curses.cpython-36m-darwin.so, 2): Library not loaded: @rpath/libncursesw.5.dylib
  Referenced from: /Users/travis/miniconda3/conda-bld/ncurses_1508961917849/_t_env/lib/python3.6/lib-dynload/_curses.cpython-36m-darwin.so
  Reason: image not found
```

**Comment by isuruf** [comment]
Yes, see last commit message.

**Comment by ocefpaf** [comment]
Thanks @isuruf! Missed that, sorry.
Closing this and re-submitting in another PR.

**Pull request closed without merging**